### ISSUE TITLE
Optimize `estimateTxDependencies` method

### ISSUE THREAD

**Issue by Torres-ssf** [FuelLabs/fuels-ts#1763]
The [`estimateTxDependencies`](https://github.com/FuelLabs/fuels-ts/blob/a2b1a545b66adf9424bbd3be049025f65e4f7a8b/packages/account/src/providers/provider.ts#L682) automates the addition of missing [OutputVariables](https://specs.fuel.network/master/tx-format/output.html#outputvariable), essential for Sway's [transfer](https://fuellabs.github.io/sway/master/reference/documentation/operations/asset/transfer/index.html?highlight=transfer#transfer) or [mint](https://fuellabs.github.io/sway/master/reference/documentation/operations/asset/mint/mint.html?highlight=mint#mint) functions.

With this, users don't need to specify the `variableOutputs` parameter: 

```ts
const { transactionResult } = await contract
  .functions
  .mint_to_addresses(address, subId, mintAmount)
  .txParams({
    variableOutputs: 3, // <—————————— this
  })
  .call();
```

> [!WARNING]
> While providing a nice DX, this approach may lead to multiple dry-run calls, potentially slowing down the transaction execution and representing more use of resources, which translates into more costs.

How it works:
1. The method performs dry runs, checking output receipts for a `Revert` with `val` equal to `TransferToAddressFailed`. This error indicates a missing `OutputVariable`.
1. If such a `Revert` receipt is detected, an `OutputVariable` is added, and another dry run is executed. This loop continues until no related revert receipt is found.

Consider a contract call minting assets to 3 addresses:

```rs
    fn mint_to_addresses(addresses: [Address; 3], sub_id: b256, mint_amount: u64) {
        assert(sub_id == TOKEN_1 || sub_id == TOKEN_2 || sub_id == TOKEN_3);
        let mut counter = 0;
        while counter < 3 {
            mint_to_address(addresses[counter], sub_id, mint_amount);
            counter = counter + 1;
        }
    }
```
Outcome:
 1. Since this function mints a coin to **3x** addresses, it will need:
    - **3x** `OutputVariable`
 1. This means the dry-run call will fail:
    - **3x** times
 1. And succeed only in the:
    - **4th** execution
 1. This will result in:
    - **4x** total dry-runs

Proposed change:
- Discontinue automatic addition of `OutputVariable`
  - Shift this responsibility to users
  - Save up resources
- Docs
  - Document how to use `OutputVariable` in the SDK documentation.
  - Include information about the error returned when an `OutputVariable` is missing

----

Related:
 - https://github.com/FuelLabs/fuels-ts/issues/925

**Comment by nedsalk** [comment]
Related (same?): #1536

**Comment by Dhaiwat10** [comment]
Does this still need to be fixed after #1767 ?

**Comment by Torres-ssf** [comment]
@Dhaiwat10 If I am not mistaken, for this one we are waiting for a possible update on `fuel-core` that will include on the `dryRun` response the amount of `OutputVariables` required by a TX when calling `dryRun`.

**Comment by arboleya** [comment]
@Torres-ssf Did we communicate this need to someone?

Ideally, we should mark this as blocked and link it with a descriptive issue on the `fuel-core` repo.

**Comment by Torres-ssf** [comment]
@arboleya @Dhaiwat10 I reckon this issue is blocked by https://github.com/FuelLabs/fuel-vm/issues/691

It seems the plan to implement this is only after mainnet.

**Comment by arboleya** [comment]
That was the missing issue, then.

Thanks for pointing that out.

**Comment by nedsalk** [comment]
Based on the issue's proposed changes (specifically "Discontinue automatic addition of `OutputVariable`"), this doesn't seem like blocked. This issue is concerned with not running transaction dependency estimation by default anymore, whereas the linked fuel-vm issue - if I understood it correctly - is concerned with getting all the dependencies in one call instead of requiring multiple calls:

> The FuelVM may support dry run mode, in which all validity rules will not abort the execution immediately. Instead, we will keep a record of what rules are failing and provide feedback at the end of the execution with all problems.

We can already make dependency estimation and opt-in thing, it'd just break the apps of our consumers that didn't specify the appropriate outputs.

**Comment by Torres-ssf** [comment]
Related to https://github.com/FuelLabs/fuel-vm/issues/691

**Comment by Torres-ssf** [comment]
@nedsalk This is blocked. 

The desired behavior, as outlined in [this issue](https://github.com/FuelLabs/fuel-vm/issues/691), is for the dry-run to indicate the number of variable outputs required for the transaction to be processed.

Implementing this feature will eliminate the need for multiple dry-runs to determine the necessary outputs.

We cannot expect users to always provide the correct number of variable outputs, especially for complex transactions involving multiple parties ( wallet connectors ).

**Comment by arboleya** [comment]
@nedsalk Agreed. I think this can be implemented already, independently from `fuel-core`.

**Comment by petertonysmith94** [comment]
Moved this back into the backlog and have lowered the priority of this one.<br><br>After exploring the outlined solution, I believe the problem to need further thought before committing to a solution. <br><br>Consider the following contract:

```
fn mint_to_addresses(condtion: bool, addresses: [Address; 3], sub_id: b256, mint_amount: u64) {
  assert(sub_id == TOKEN_1 || sub_id == TOKEN_2 || sub_id == TOKEN_3);  
  
  // 1 Variable Output  
  if (condition) {
    mint_to_address(addresses[0], sub_id, mint_amount);
  }
  // 3 Variable Outputs
  else {
    let mut counter = 0;
    while counter < 3 {
      mint_to_address(addresses[counter], sub_id, mint_amount);
      counter = counter + 1;
    }
  }
}
```

This could either have 1 or 3 variable outputs based upon some condition.

**Comment by Torres-ssf** [comment]
This was solved with the [assemble_tx](https://github.com/FuelLabs/fuels-ts/pull/3747) query
